packages/x-data-grid/src/components/grid/export/GridExportTrigger.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { ButtonProps } from '@mui/material/Button';
+import { useGridApiContext } from '../../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../../hooks/utils/useGridRootProps';
+import {
+  useGridComponentRenderer,
+  RenderProp,
+} from '../../../hooks/utils/useGridComponentRenderer';
+
+export interface GridExportTriggerState {}
+
+export interface GridExportTriggerProps extends ButtonProps {
+  exportType: 'csv' | 'excel' | 'print';
+  render?: RenderProp<GridExportTriggerState>;
+}
```

Comment:
This will be slightly more complex: since we don't call methods directly, we need to pass export options to them.
We can use discriminated unions to type them appropriately:

```ts
export type GridExportTriggerProps = ButtonProps & {
  render?: RenderProp<GridExportTriggerState>;
} & (
    | {
        exportType: 'csv';
        exportOptions?: {
          delimiter?: string;
        };
      }
    | {
        exportType: 'print';
        exportOptions?: {
          pageStyle?: string;
        };
      }
    | {
        exportType: 'excel';
        exportOptions?: {
          exceljsPreProcess?: () => void;
        };
      }
  );
```

Alternatively, we can have separate trigger components for each export type. In this specific case, this might be helpful for isolation between packages – since Excel export is a Premium feature, the Excel export trigger will only be exported in the Premium package.